core/modules/entity/entity.module

```diff
@@ -350,8 +350,14 @@ function entity_extract_ids($entity_type, $entity) {
   $info = entity_get_info($entity_type);
 
   // Objects being created might not have id/vid yet.
+  if (!empty($info['entity keys'])) {
   $id = isset($entity->{$info['entity keys']['id']}) ? $entity->{$info['entity keys']['id']} : NULL;
   $vid = ($info['entity keys']['revision'] && isset($entity->{$info['entity keys']['revision']})) ? $entity->{$info['entity keys']['revision']} : NULL;
```

Comment:
This code looks good @philsward, but you will need to indent these 2 lines by 2 spaces, to adhere to our coding standards.